src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -870,7 +870,7 @@ protected virtual void ReportUnassigned(Symbol symbol, SyntaxNode node)
                 // We've already reported the use of a local before its declaration.  No need to emit
                 // another diagnostic for the same issue.
             }
-            else if (!_alreadyReported[slot])
+            else if (!_alreadyReported[slot] && VariableType(symbol)?.IsErrorType() != true)
```

Comment:
This results in the suppression of cascaded diagnostics, as you can see in the changed tests. Situations in which we fail to infer a type also used to result in cascaded diagnostics.

---
In reply to: [75760660](https://github.com/dotnet/roslyn/pull/13274#discussion_r75760660) [](ancestors = 75760660)